authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/pages/connectormanagement/ConnectorManagementApiService.kt

```diff
@@ -180,95 +185,139 @@ class ConnectorManagementApiService(
-            Log.error("Connector with this client ID already exists. connectorId=$connectorId, customerOrganizationId=$customerOrganizationId, userId=$userId, clientId=$clientId.")
-            return CreateConnectorResponse.error("Connector with this client ID already exists.", timeUtils.now())
+        existingConnector.apply {
+            frontendUrl = connectorParams.frontendUrl
+            endpointUrl = connectorParams.endpointUrl
+            managementUrl = connectorParams.managementUrl
+            jwksUrl = connectorParams.jwksUrl
+            type = ConnectorType.PROVIDED
+            update()
         }
 
-        connectorService.createProvidedConnector(
+        registerConnectorAtDaps(
+            clientId = existingConnector.clientId,
             connectorId = connectorId,
-            organizationId = customerOrganizationId,
-            providerOrganizationId = providerOrganizationId,
-            environment = deploymentEnvId,
-            clientId = clientId,
-            createConnectorParams = connector,
-            createdBy = userId
+            certificate = connectorParams.certificate,
+            jwksUrl = connectorParams.jwksUrl,
+            deploymentEnvId = deploymentEnvId
         )
-        registerConnectorAtDaps(clientId, connectorId, connector, deploymentEnvId)
 
-        Log.info("Connector for foreign organization registered. connectorId=$connectorId, customerOrganizationId=$customerOrganizationId, userId=$userId.")
-        return CreateConnectorResponse.ok(connectorId, clientId, timeUtils.now())
+        Log.info("Connector for foreign organization configured and registered in DAPS. connectorId=$connectorId, customerOrganizationId=${existingConnector.organizationId}, userId=$userId.")
+        return CreateConnectorResponse.ok(connectorId, existingConnector.clientId, timeUtils.now())
+    }
+
+    private fun assertProvidedConnectorConfigurable(connector: ConnectorRecord, userOrganizationId: String) {
+        if (connector.providerOrganizationId != userOrganizationId) {
+            Log.error("User is not allowed to configure this connector. connectorId=${connector.connectorId}, userOrganizationId=$userOrganizationId.")
+            unauthorized("User is not allowed to configure this connector.")
+        }
+
+        if (connector.type != ConnectorType.CONFIGURING) {
+            Log.error("Connector is not in configuring state. connectorId=${connector.connectorId}.")
+            unauthorized("Connector is not in configuring state.")
+        }
     }
 
     private fun isValidUrlConfiguration(
-        frontendUrlString: String,
-        endpointUrlString: String,
-        managementUrlString: String
+        frontendUrlString: String?,
+        endpointUrlString: String?,
+        managementUrlString: String?
     ): Boolean {
+        if (frontendUrlString == null || endpointUrlString == null || managementUrlString == null) {
+            return false
+        }
+
```

Comment:
try {
            return listOf(frontendUrlString, endpointUrlString, managementUrlString)
                .map { URL(it).protocol }
                .all { it == "https" }
        } catch (e: MalformedURLException) {
            return false
        }

Reply:
This is a side-effect of my DRYing some stuff. I am reusing the same ConnectorParams object for all connector types, but for configuring connectors (and for new CaaS), the URLs are null. While I'm not calling this function in those cases anyway, I get complains when calling it elsewhere because ConnectorParams.xUrl is nullable.

I could have just asserted non-null by using `!!` but I figured checking if the URL is null here is more clean, since it would help catching bugs with the URL somehow not being set